1-Draft/RFC00XX-Native-Command-Error-Handling.md

```diff
@@ -0,0 +1,229 @@
+  length of the output stream is non-zero. This is the same as existing PowerShell treatment of this
+  case.
+- `$true`: the boolean value of a native command is it's exit status (`$?`), and
+
+This would allow syntax like `if` and `while` to be usefully combined with native commands.
+
+#### Add strict pipeline chain failure semantics.
+
+Treat only ignored exit statuses as exceptions.
+
+The `$PSStrictPipeLineChain` preference variable would govern the exit status in the last command of
+a pipeline. Possible values would be:
+
+- `$false`: (the default) would ignore `$false` exit status on the last command. This is the same as
+  existing PowerShell treatment of this case.
+- `$true`: for a pipeline that is being used in a conditional context (see above), and where the
+  exist status of the last command in the pipeline is `$false`, would create a terminating error
+  from the non-terminating errors in the command error stream.
+
+This should improve on the basic specification by allowing the idiom to be usefully combined with
+pipeline chaining.
+
+### Use dynamic scope/Set-StrictMode
+
+This approach implements dynamic scoping for native command error management using a cmdlet.
+
+Some of the above would be enabled with `Set-StrictMode -version 6` instead of with boolean
+preference variables.
+
+The dynamic scoping approach would improve on earlier listed approaches by limiting the scope of
+error handling configuration so that script functions could not have an effect on the error handling
+mode in the calling scope.
+
+If `Set-StrictMode` is used for this, it would need to enable some combination of enhancements that
+will not raise errors on scripts that have already implemented strong native command error handling.
+
+### Use lexical scope/Exception handling extensions
+
+Dynamic scoping approach would have side-effects on called scripts. This is analogous to the
+behavior of Bourne type shells, which maintain this behavior for
+[historic compatibility](http://austingroupbugs.net/view.php?id=52) reasons.
+
+Lexical scoping of native command error handling would improve on earlier options by integrating
+native command error handling fully into the existing exception handling without any
+side-effects in calling or called scripts.
+
+With this approach, native command error handling mode would be used through language syntax instead
+of preference variables or cmdlets. A possible syntax might be to add a strictness option to the try
+statement which applies to the lexical scope of the try statement.
+
+Lexical scope requires more runtime overhead; a mapping must be maintained at runtime between each
+runtime scope and its corresponding lexical scope that existed at parse-time. In addition, lexical
+scope may not apply well to errors. Errors produce stack traces at runtime and should be handled in
+a runtime-facing way. For these reasons, this alternative is not under consideration.
+
```

Comment:
some of these would need a _bunch_ more explanation in order to pursue. I'm not sure what to do with this section. My assumption is that these aren't really part of the spec, just forestalling "what if" conversations?